armi/reactor/converters/axialExpansion/assemblyAxialLinkage.py

```diff
@@ -0,0 +1,220 @@
+                single=True,
+            )
+        if (
+            lstLinkedC[1] is None
+            and self.linkedBlocks[b][1] is not None
+            and not self.linkedBlocks[b][1].hasFlags(Flags.DUMMY)
+        ):
+            # only print debug is there is a linked block above in the first place,
+            # and if that linked block is not the DUMMY block
+            runLog.debug(
+                f"Assembly {self.a}, Block {b}, Component {c} has nothing linked above it!",
+                single=True,
+            )
+
+    @staticmethod
+    def _determineLinked(componentA, componentB):
+        """Determine axial component linkage for two components.
+
+        Parameters
+        ----------
+        componentA : :py:class:`Component <armi.reactor.components.component.Component>`
+            component of interest
+        componentB : :py:class:`Component <armi.reactor.components.component.Component>`
+            component to compare and see if is linked to componentA
+
+        Notes
+        -----
+        - Requires that shapes have the getCircleInnerDiameter and getBoundingCircleOuterDiameter defined
+        - For axial linkage to be True, components MUST be solids, the same Component Class, multiplicity, and meet
+        inner and outer diameter requirements.
+        - When component dimensions are retrieved, cold=True to ensure that dimensions are evaluated
+        at cold/input temperatures. At temperature, solid-solid interfaces in ARMI may produce
+        slight overlaps due to thermal expansion. Handling these potential overlaps are out of scope.
+
+        Returns
+        -------
+        linked : bool
+            status is componentA and componentB are axially linked to one another
+        """
+        if (
+            (componentA.containsSolidMaterial() and componentB.containsSolidMaterial())
+            and isinstance(componentA, type(componentB))
+            and (componentA.getDimension("mult") == componentB.getDimension("mult"))
+        ):
+            if isinstance(componentA, UnshapedComponent):
+                runLog.warning(
+                    f"Components {componentA} and {componentB} are UnshapedComponents "
+                    "and do not have 'getCircleInnerDiameter' or getBoundingCircleOuterDiameter methods; "
+                    "nor is it physical to do so. Instead of crashing and raising an error, "
+                    "they are going to be assumed to not be linked.",
+                    single=True,
+                )
+                linked = False
+            else:
+                idA, odA = (
+                    componentA.getCircleInnerDiameter(cold=True),
+                    componentA.getBoundingCircleOuterDiameter(cold=True),
+                )
+                idB, odB = (
+                    componentB.getCircleInnerDiameter(cold=True),
+                    componentB.getBoundingCircleOuterDiameter(cold=True),
+                )
```

Comment:
```suggestion
                idA = componentA.getCircleInnerDiameter(cold=True)
                odA = componentA.getBoundingCircleOuterDiameter(cold=True)
                idB = componentB.getCircleInnerDiameter(cold=True)
                odB = componentB.getBoundingCircleOuterDiameter(cold=True)
```

Reply:
While you _can_ define two different variables on one line, I don't think it's easier to read.

On the one hand, this is just my personal preference, so I guess I won't die on this hill. On the other hand, this is something we try to studiously avoid in ARMI.